sp/src/game/server/hl2/npc_zombine.cpp

```diff
@@ -1802,6 +1845,380 @@ void CNPC_MetroZombie::ReadyGrenadeSound( void )
+		PrecacheScriptSound( "HEVGlowbie.Vox_Flatline" );
+		PrecacheScriptSound( "HEVGlowbie.Vox_Pain" );
+
+		PrecacheScriptSound( "ATV_engine_null" );
+		PrecacheScriptSound( "HEVGlowbie.Charge" );
+		PrecacheScriptSound( "Glowbie.Attack" );
+		break;
+	case EZ_VARIANT_XEN:
+		modelVariant = "xenbie";
+		PrecacheScriptSound( "Xenbie.FootstepRight" );
+		PrecacheScriptSound( "Xenbie.FootstepLeft" );
+		PrecacheScriptSound( "HEVXenbie.ScuffRight" );
+		PrecacheScriptSound( "HEVXenbie.ScuffLeft" );
+		PrecacheScriptSound( "Xenbie.AttackHit" );
+		PrecacheScriptSound( "Xenbie.AttackMiss" );
+		PrecacheScriptSound( "HEVXenbie.Pain" );
+		PrecacheScriptSound( "HEVXenbie.Die" );
+		PrecacheScriptSound( "HEVXenbie.Alert" );
+		PrecacheScriptSound( "HEVXenbie.Idle" );
+		PrecacheScriptSound( "HEVXenbie.ReadyGrenade" );
+		PrecacheScriptSound( "HEVXenbie.Vox_Idle" );
+		PrecacheScriptSound( "HEVXenbie.Vox_Alert" );
+		PrecacheScriptSound( "HEVXenbie.Vox_Charge" );
+		PrecacheScriptSound( "HEVXenbie.Vox_Flatline" );
+		PrecacheScriptSound( "HEVXenbie.Vox_Pain" );
+
+		PrecacheScriptSound( "ATV_engine_null" );
+		PrecacheScriptSound( "HEVXenbie.Charge" );
+		PrecacheScriptSound( "Xenbie.Attack" );
+		break;
+	default:
+		modelVariant = "zombie";
+		PrecacheScriptSound( "Zombie.FootstepRight" );
+		PrecacheScriptSound( "Zombie.FootstepLeft" );
+		PrecacheScriptSound( "HEVZombie.ScuffRight" );
+		PrecacheScriptSound( "HEVZombie.ScuffLeft" );
+		PrecacheScriptSound( "Zombie.AttackHit" );
+		PrecacheScriptSound( "Zombie.AttackMiss" );
+		PrecacheScriptSound( "HEVZombie.Pain" );
+		PrecacheScriptSound( "HEVZombie.Die" );
+		PrecacheScriptSound( "HEVZombie.Alert" );
+		PrecacheScriptSound( "HEVZombie.Idle" );
+		PrecacheScriptSound( "HEVZombie.ReadyGrenade" );
+		PrecacheScriptSound( "HEVZombie.Vox_Idle" );
+		PrecacheScriptSound( "HEVZombie.Vox_Alert" );
+		PrecacheScriptSound( "HEVZombie.Vox_Charge" );
+		PrecacheScriptSound( "HEVZombie.Vox_Flatline" );
+		PrecacheScriptSound( "HEVZombie.Vox_Pain" );
+
+		PrecacheScriptSound( "ATV_engine_null" );
+		PrecacheScriptSound( "HEVZombie.Charge" );
+		PrecacheScriptSound( "Zombie.Attack" );
+		break;
+	}
+	if (GetModelName() == NULL_STRING)
+	{
+		SetModelName( AllocPooledString( UTIL_VarArgs( "models/zombie/%s_hev.mdl", modelVariant ) ) );
+	}
+	if (GetTorsoModelName() == NULL_STRING && m_tEzVariant != EZ_VARIANT_XEN)
+	{
+		SetTorsoModelName( AllocPooledString( UTIL_VarArgs( "models/zombie/%s_hev_torso.mdl", modelVariant ) ) );
+	}
+	if (GetLegsModelName() == NULL_STRING && m_tEzVariant != EZ_VARIANT_XEN)
+	{
+		SetLegsModelName( AllocPooledString( UTIL_VarArgs( "models/zombie/%_hev_legs.mdl", modelVariant ) ) );
+	}
+
+	PrecacheModel( STRING( GetModelName() ) );
+
+	BaseClass::Precache();
+}
+
+void CNPC_HEVZombie::ChooseDefaultGrenadeType()
+{
+	int chance = random->RandomInt( 0, 3 );
+
+	// If we set the type to "Arbeit", give them Xen grenades
+	if (m_tEzVariant == EZ_VARIANT_ARBEIT)
```

Comment:
Shouldn't the rad variant also be given Xen grenades? 🤔

Reply:
Perhaps... That seems like a good idea to me